enterprise/views.py

```diff
@@ -2339,6 +2341,20 @@ def get(self, request, *args, **kwargs):
 
         # Enrollments through Cornerstone have some params in querystring, need to store those params if exists.
         if course_key:
+            session_token = request.GET.get('sessionToken')
+            if session_token:
+                csod_customer_configuration_model = apps.get_model(
+                    'integrated_channel',
+                    'CornerstoneEnterpriseCustomerConfiguration'
+                )
+                with transaction.atomic():
```

Comment:
if you want locking in addition to this transaction, you'll need a `select_for_update` like we do in Blackboard https://github.com/edx/edx-enterprise/blob/master/integrated_channels/blackboard/client.py#L368